nncf/utils.py

```diff
@@ -358,3 +359,23 @@ def training_mode_switcher(model: torch.nn.Module, is_training: bool = True):
         yield
     finally:
         model.train(is_original_mode_training)
+
+
+def compute_FLOPs_hook(module, input_, output, dict_to_save, name):
+    if isinstance(module, (nn.Conv1d, nn.ConvTranspose1d)):
+        ks = module.weight.data.shape
+        mac_count = ks[0] * ks[1] * ks[2] * output.shape[2]
+    elif isinstance(module, (nn.Conv2d, nn.ConvTranspose2d)):
+        ks = module.weight.data.shape
+        mac_count = ks[0] * ks[1] * ks[2] * ks[3] * output.shape[2] * output.shape[3]
+    elif isinstance(module, (nn.Conv3d, nn.ConvTranspose3d)):
+        ks = module.weight.data.shape
+        mac_count = ks[0] * ks[1] * ks[2] * ks[3] * ks[4] * output.shape[2] * \
+                    output.shape[3] * output.shape[4]
```

Comment:
seems like you can shorten it to `np.prod(ks) * np.prod(output.shape[2:])` for convolutions of any dimension

Reply:
also I guess you should include bias addition (in case it's present) to the FLOPs count to be exact (same for modules of `Linear` type).

Reply:
The function was shortened and bias addition was added.